app/models/hud_reports/report_cell.rb

```diff
@@ -53,11 +55,19 @@ def user
     end
 
     def members
-      @members ||= join_universe
+      @members ||= if report_instance.artifacts_stored?
+        load_members_from_s3
+      else
+        join_universe
+      end
     end
 
     def count
-      @count ||= universe_members.count
+      @count ||= if report_instance.artifacts_stored?
+        count_from_s3
+      else
+        universe_members.count
+      end
```

Comment:
I'm probably missing something, but I think this could just be:

```suggestion
      @count ||= if report_instance.artifacts_stored?
        members.count
      else
        universe_members.count
      end
```

Since both methods are loading the data from S3.  That way it would only have to be loaded once, and then it would just pull from memory on the next request.